docs/adr/0007-unit-testing-framework.md

```diff
@@ -0,0 +1,74 @@
+# Unit Testing Framework
+
+- Status: proposed
+  - Supercedees E-compatible unit test frameworks (#0001)
+- Deciders: Monitor team
+- Date: 2023-06-14
+
+Technical Story: [description | ticket/issue URL] <!-- optional -->
+
+## Context and Problem Statement
+
+With the transition to Next.js/React, the team wants to consider testing frameworks again, to confirm what works best for the new architecture.
```

Comment:
We were discussing this elsewhere, I though @Vinnl had a good take on this (paraphrasing): we should aim for 100% code coverage, with files/lines temporarily disabled if they are too difficult (or unimportant e.g. deprecated code) to get to immediately.

Reply:
💯! 😍 
Well, 100% is better than our current 19%, so I like that trend.
I'm just happy if we can officially document that 100% or 95% or even 90% aspirational life goal in this ADR and then eventually set up CI to fail if it drops below our threshold. Since I think we know that if we just land code now and then write the tests after we launch, those tests+coverage rarely get added.

Reply:
Yeah, my preferred strategy would be to just set the coverage threshold to 100% right away, including CI failure, and then ignore all existing files for coverage. That way, all new files can have active consideration of what needs to be covered and what doesn't.

I don't think we need an ADR for that though.

Reply:
> I don't think we need an ADR for that though.

Agreed, I don't think we need a separate code coverage ADR, I was just hoping for a mention of 90-100% code coverage in this ADR so I can hold everybody accountable to that number. 🙏

Reply:
But Peter, I don't want you to have to hold people accountable to that - we have robots for that! https://github.com/mozilla/blurts-server/pull/3274